### DISCUSSION TITLE
k3s ctr images ls fails as non root user

### DISCUSSION THREAD

**Discussion by nithinj** [k3s-io/k3s#7801] [Q&A] [answered]
```
k3s ctr images ls
ctr: failed to dial "/run/k3s/containerd/containerd.sock": connection error: desc = "transport: error while dialing: dial unix /run/k3s/containerd/containerd.sock: connect: permission denied"

sudo ls -l /run/k3s/containerd/containerd.sock
srw-rw---- 1 root containerd 0 Jun 19 11:53 /run/k3s/containerd/containerd.sock

groups $USER
nithinj : nithinj sudo docker containerd

k3s -v
k3s version v1.26.5+k3s1 (7cefebea)
go version go1.19.9

k3s ctr -v
ctr github.com/k3s-io/containerd v1.7.1-k3s1

```
I did `newgrp containerd` before firing the failed command

**Comment by brandond** [comment, marked as answer]
You generally have to log out and back in after changing your group membership, even if groups shows the changes. If that doesn't do it, check the permissions of the parent directories.

K3s itself is not responsible for enforcing any filesystem access controls, this is all just standard Linux stuff.

**Reply by nithinj** [reply]
I have tried logout and login without any luck. My parent dirs are having access to the group.
```
sudo ls -ld /run/k3s/containerd
drwxrwx--x 5 root containerd 140 Jun 19 11:53 /run/k3s/containerd
```
Thanks for the update. I will post if there is any updates on this.

**Reply by nithinj** [reply]
well, a restart of the system worked. As you suggested it could be a hiccup in the file system.
